nmrfx-core/src/main/java/org/nmrfx/chemistry/AtomEnergyProp.java

```diff
@@ -106,35 +105,35 @@ public static void readPropFile(String fileName) throws FileNotFoundException, I
             bf = new BufferedReader(new FileReader(fileName));
         }
 
-        lineReader = new LineNumberReader(bf);
         List<String> headerS = Arrays.asList();
         //AtomType        HardRadius      RMin    E       Mass    HBondType
-
-        while (true) {
-            string = lineReader.readLine();
-            if (string != null) {
-                List<String> stringS = Arrays.asList(string.split("\\s+"));
-                if (string.startsWith("AtomType")) {
-                    headerS = stringS;
+        try (LineNumberReader lineReader = new LineNumberReader(bf)) {
+            while (true) {
+                string = lineReader.readLine();
+                if (string != null) {
```

Comment:
This could probably be rewritten as something like
```
while((string = lineReader.readLine()) != null) {
..
}
```
to avoid the infinite loop + break on null.
This pattern was common when using BufferedReader to read lines.

Nowadays, we'd use `Files.lines(..)` and stream over it instead.
Here, LineNumberReader is used, but never to get the last read line number, so it could be replaced.

Reply:
Wasn't able to use Files.lines(..) here, as when I tried running from the jar file, I couldn't get the path right, (BufferedReader was being created with getResourceAsStream) so removed LineReader since redundant and just used buffered reader with readLine in the while condition